src/array/cpu/array_utils.h

```diff
@@ -39,13 +39,12 @@ class IdHashMap {
   void Update(IdArray ids) {
     const IdType* ids_data = static_cast<IdType*>(ids->data);
     const int64_t len = ids->shape[0];
-    IdType newid = oldv2newv_.size();
     for (int64_t i = 0; i < len; ++i) {
       const IdType id = ids_data[i];
-      if (!Contains(id)) {
-        oldv2newv_[id] = newid++;
-        filter_[id & kFilterMask] = true;
-      }
+      // std::unorderd_map::insert assures that an insertion will not happen if the
```

Comment:
unordered_map.

Also I think the membership check of `Contains(id)` for absent IDs should be generally faster than the one in `std::unordered_map`.  That's why we had this check in the first place.

Not sure how the speed gain was though.  Maybe @zheng-da could give some more insights on this?

Reply:
It's a very subtle difference. Previous codes could have two look ups (one in `Contains`, one in `oldv2new_[id]=newid++`, while this one only has one look up (in std::unordered_map::insert).

Reply:
Make sense to me.